WooCommerce/UITestsFoundation/Screens/Products/ProductsScreen.swift

```diff
@@ -47,8 +49,26 @@ public class ProductsScreen: ScreenObject {
     }
 
     @discardableResult
-    public func selectProduct(atIndex index: Int) throws -> SingleProductScreen {
-        XCUIApplication().tables.cells.element(boundBy: index).tap()
+    public func selectProductByIndex(atIndex index: Int) throws -> SingleProductScreen {
+        app.tables.cells.element(boundBy: index).tap()
         return try SingleProductScreen()
     }
+
+    public func selectProductByName(name: String) throws -> SingleProductScreen {
+        app.tables.cells.staticTexts[name].tap()
+        return try SingleProductScreen()
+    }
```

Comment:
Swift allows having multiple functions with the same name, but different parameters. Also, it allows providing an "external" parameter name, and the "internal" one (this is done in current version of `selectProductByIndex `: when the method is called, the parameter is listed as `atIndex`, however inside the functions it uses just `index`, for a better legibility).

Wile keeping in mind that Swift forces to provide a parameter name during the call (unless the "external" name is `_`), the code in its current state looks somewhat redundant:

```swift
selectProductByName(name: firstProductName)
```

IMO we can shorten the names and make them more consistent by utilizing a descriptive naming approach:

```suggestion
    public func selectProduct(byIndex index: Int) throws -> Self {
        app.tables.cells.element(boundBy: index).tap()
        return self
    }

    public func selectProduct(byName name: String) throws -> Self {
        app.tables.cells.staticTexts[name].tap()
        return self
    }
```

wdyt?